src/main/java/duke/command/Duke.java

```diff
@@ -0,0 +1,153 @@
+
+        if (userCommands.get(0).equals("list")) {
+            displayTaskList();
+        } else if (userCommands.get(0).equals("done")) {
+            markCompletedTasks(userCommands.subList(1, userCommands.size()));
+        } else {
+            try {
+                appendNewTask(userCommands);
+            } catch (DukeExceptions e) {
+                System.out.println("Invalid instructions detected in the following line: " + line);
+                System.out.println("Clarification on the use of this app can be found ...");
+                System.out.println(lineSpacing);
+            }
+        }
+    }
+
+    public static void displayTaskList() {
+        int counter = 0;
+        System.out.println(lineSpacing);
+        System.out.println("\tHere are the tasks in your list:");
+        while (counter < tasks.size()) {
+            Task task = tasks.get(counter);
+            System.out.print("\t" + (counter+1) + ".");
+            task.printTask();
+            counter ++;
+        }
+        System.out.println(lineSpacing);
+    }
+
+    public static void markCompletedTasks(List<String> taskIndexes) {
+        System.out.println(lineSpacing);
+        System.out.println("\tNice! I've marked this task as done:");
+
+        for (String index: taskIndexes) {
+            int indexInt;
+            try {
+                indexInt = Integer.parseInt(index) - 1;
+                if (indexInt >= tasks.size()){
+                    throw new DukeExceptions();
+                }
+                else {
+                    tasks.get(indexInt).setIsCompleted(true);
+                    System.out.print("\t");
+                    tasks.get(indexInt).printTask();
+                }
+            } catch (DukeExceptions e) {
+                System.out.println("\tTask number " + index + " does not exist");
+                System.out.println(lineSpacing);
+            } catch (NumberFormatException e) {
+                System.out.println("\tInvalid Expression! Must provide task numbers (" + index + ")");
+                System.out.println(lineSpacing);
+            }
+        }
+        System.out.println(lineSpacing);
+    }
+
+    public static void appendNewTask(List<String> userCommands) throws DukeExceptions{
+        System.out.println(lineSpacing);
+        switch (userCommands.get(0)) {
+        case "todo":
+            List<String> taskNameTodo = userCommands.subList(1, userCommands.size());
+            tasks.add(new ToDo(
+                    String.join(" ", taskNameTodo),
+                    userCommands.get(0)
+            ));
+            System.out.println("\tGot it. I've added this task: ");
+            break;
+        case "deadline":
+            int indexDeadline = userCommands.indexOf("/by");
+            if (indexDeadline == -1) {
+                throw new DukeExceptions();
+            }
+            List<String> taskNameDeadline = userCommands.subList(1, indexDeadline);
```

Comment:
The name of the variable is a bit confusing. It should represent "the name of the deadline". So maybe "deadlineName" is a better option?